File: Week 2/Problem Set/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int get_total_letters(string text);
int get_total_words(string text);
int get_total_sentences(string text);
float get_grade(float letters, float words, float sentences);

int main(void)
{
    string text = get_string("Text: ");

    float letters = get_total_letters(text);
    float words = get_total_words(text);
    float sentences = get_total_sentences(text);

    float grade = get_grade(letters, words, sentences);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }

    printf("Grade %.0f\n", grade);
}

int get_total_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }

    return letters;
}

int get_total_words(string text)
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words += 1;
        }
    }

    return words;
}

int get_total_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] == '!') || (text[i] == '?') || (text[i] == '.'))
        {
            sentences += 1;
        }
    }

    return sentences;
}

float get_grade(float letters, float words, float sentences)
{
    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    float grade = (0.0588 * L) - (0.296 * S) - 15.8;

    return grade;
}

File: Week 6/Problem Set/readability.py
def get_grade(letters: int, words: int, sentences: int):
    L = (letters / words) * 100
    S = (sentences / words) * 100
    grade = (0.0588 * L) - (0.296 * S) - 15.8

    return grade


def get_total_sentences(text: str):
    sentences = 0
    for c in text:
        for i in ["!", "?", "."]:
            if i == c:
                sentences += 1

    return sentences


def get_total_words(text: str):
    words = 1
    for c in text:
        if c.isspace():
            words += 1

    return words


def get_total_letters(text: str):
    letters = 0
    for c in text:
        if c.isalpha():
            letters += 1

    return letters


def main():
    text = str(input("Text: "))

    letters = get_total_letters(text)
    words = get_total_words(text)
    sentences = get_total_sentences(text)

    grade = get_grade(letters, words, sentences)

    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")

    print(f"Grade {round(grade)}")


if __name__ == "__main__":
    main()
